core/src/main/java/com/google/instrumentation/trace/Trace.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.trace;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.instrumentation.common.Provider;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/** Class that manages a global instance of the {@link TraceService}. */
```

Comment:
Yes objects are safe to cache and encouraged to be used as Logger:
private static final Tracer tracer = Trace.getTracer();

A lot of blog-posts about why singletons are "evil". An alternative of this would be to manually propagate an instance of a TraceService within your code, which seems much worse from an integration point of view.

The reason why we went the path of reflection loading is that we expect users to do 0 initialization (just link the implementation to their binary), or minimum initialization work.